probe_src/probe_py/analysis.py

```diff
@@ -1,53 +1,64 @@
 import typing
-import networkx
-from .parse_probe_log import ProvLog, Op, CloneOp, ExecOp, WaitOp, CLONE_THREAD
+import networkx as nx
+from .parse_probe_log import ProcessTreeProvLog, Op, CloneOp, ExecOp, WaitOp, CLONE_THREAD
 
-
-def construct_process_graph(process_tree_prov_log: ProvLog) -> str:
+def construct_process_graph(process_tree_prov_log: ProcessTreeProvLog):
     """
     Construct a happens-before graph from process_tree_prov_log
 
     The graph shows the order that prov ops happen.
     """
-    Node: typing.TypeAlias = tuple[int, int, int, Op | str]
```

Comment:
Can we take out `time` and `node_count`?

I initially had time (the time the process was started) because Linux can reuse PIDs after one dies, but now I think the chance of that happening is so remote, we can disregard it.

I believe in having no "extra" information, and I think `node_count` is extra in the sense that we don't need it for anything. We can already locate the op with PID, Ex ID, TID, Op no.

Reply:
- Removed `_time`
- But we do need node_count to identify a particular node while building the graph. node_count is not part of the graph's label; instead, it is used to identify nodes so that we can establish edges after adding the nodes to the graph.